micro-rdk/src/native/tls.rs

```diff
@@ -78,14 +78,20 @@ impl NativeTlsStream {
                     .iter()
                     .map(|c| rustls::Certificate(c.clone()))
                     .collect();
+            let priv_keys: Vec<_> =
+                rustls_pemfile::rsa_private_keys(&mut BufReader::new(tls_cfg.srv_key.as_slice()))
+                    .unwrap()
+                    .iter()
+                    .map(|k| rustls::PrivateKey(k.clone()))
+                    .collect();
 
             let mut cfg = ServerConfig::builder()
                 .with_safe_default_cipher_suites()
                 .with_safe_default_kx_groups()
                 .with_protocol_versions(&[&rustls::version::TLS12])
                 .map_err(|err| std::io::Error::new(std::io::ErrorKind::Other, err))?
                 .with_no_client_auth()
-                .with_single_cert(cert_chain, rustls::PrivateKey(tls_cfg.srv_key.clone()))
+                .with_single_cert(cert_chain, priv_keys[0].clone())
```

Comment:
There should probably be a check that we have one, at least. What should we be doing if we have multiple. Is there a `with_multiple_cert`? I haven't seen such a thing.

Reply:
[`with_single_cert`](https://docs.rs/rustls/latest/rustls/struct.ConfigBuilder.html#method.with_single_cert) actually takes a `Vec<CertificateDer<'static>>` (DER encoded certificates) as the first argument, `cert_chain`.

If you mean the ability to dynamically respond using specific cert-key combos, there's [`Acceptor`](https://docs.rs/rustls/latest/rustls/server/struct.Acceptor.html) for async and [`ResolvesServerCert`](https://docs.rs/rustls/latest/rustls/server/trait.ResolvesServerCert.html#) for blocking.